controller/movies.js

```diff
@@ -0,0 +1,45 @@
+const movies = require('../movies')
+const getAllMovies = (request, response) => {
+  return response.send(movies)
+}
+
+const getMovie = (request, response) => {
+  const { input } = request.params
+
+  const foundMovie = movies.filter((movie) => { // filter function is for each movie.line 10-19 will run for each movie.
+    if (movie.title.toUpperCase().includes(input.toUpperCase())) { // search for input in the title,return true if found
+      return true // if the condition is false it will enter the for loop next.If true, it will move to next movie.
+    }
+    for (let i = 0; i < movie.directors.length; i++) { // search for input in the the directors array,return true if found
+      if (movie.directors[i].toUpperCase().includes(input.toUpperCase())) {
+        return true
+      }
+    }
+
+    return false // return false if input is not in the movie title or directors array
+  })
+
+  if (foundMovie.length >= 1) {
+    return response.send(foundMovie)
+  } else {
+    return response.send('No movies found!')
```

Comment:
_Nit_ instead of returning 'No Movies Found' you want to return a status of 404 because this still gives a 200 which is a positive message and we dont want that.